### PR TITLE
Added test for allowEndTurnIf on SimpleFlow

### PR DESCRIPTION
**Pull request by karel1980**
A first stab at #91 to see if you think I'm going in the right direction.
- I'm only testing SimpleFlow. The fact that it has to be implemented both in simple flow and in phase flow is a code smell. Isn't simple flow just a special case of phase flow (where there is only one fixed phase)? Room for simplification here.

- I added a simple check to tic-tac-toe to force at least one move. The flow tests seem very low level. I had to resort to manually testing with tic tac toe to be confident that it worked.

- This is not covering everything yet. You can still specify an invalid move (e.g. clickCell(999) in tic-tac-toe, and endTurnIf will automatically end the turn.

- How should the system behave if endTurnIf and allowEndTurn contradict each other? (maybe that's what happening in tic-tac-toe right now.

### CODE REVIEW

**Comment by karel1980** [comment]
I'll get the CLA in order

**Comment by karel1980** [comment]
I signed it!

File: src/core/flow.js [hunk 1]
```diff
@@ -101,18 +101,22 @@ export function Flow({ ctx, events, init, validator, processMove }) {
  *                               Whenever `condition` is true the `action` is run.
  *                               Triggers are processed one after the other in the
  *                               order they are defined at the end of each move.
+ * @param {...object} allowEndTurnIf - Turn not allowed to end if this returns true.
```
Comment (nicolodavis) [inline src/core/flow.js]: Hmm, I think this is going to make the semantics a bit confusing. Like you mention, what happens if this conflicts with `endTurnIf`?

How about just using a simple whitelist similar to `allowedMoves`. In the flow section, you can have:

```
flow: {
  // default value if not provided.
  allowedEvents: [ 'endTurn', 'endPhase', 'pass'],
}
```

Comment (karel1980) [reply src/core/flow.js]: Yes the conflict with endTurnIf is a problem. Simply whitelisting would work.
For my use case the 'endTurn' event is only allowed after specific moves have been made,
but I can make it work by (a) disabling it using allowedEvents on flow and (b) implementing a move which sets a flag on game and (c) endTurnIf picks up the flag.

**Comment by nicolodavis** [comment]
@karel1980 As part of getting rid of `SimpleFlow` (and make it a special case of `FlowWithPhases`), I actually had to implement the event whitelist as well: https://github.com/google/boardgame.io/commit/1abc042de2ff9192a54ab15b0d1fdb8ebf3948ee

Closing this PR if that commit subsumes this (I'll release it in the next version of the NPM soon). Feel free to re-open if it doesn't fix your problem.

**Pull request closed without merging**